### ISSUE TITLE
Core Colorpicker JS doesn't work with RequireJS

### ISSUE THREAD

**Issue by PivitParkour94** [magento/magento2#17008]
### Preconditions

1. Magento 2.2.x
2. Mysql  Ver 15.1 Distrib 10.2.15-MariaDB, for Linux (x86_64) using readline 5.1
3. PHP 7.0.30
4. RequireJS 2.1.11

### Steps to reproduce

1. Create a javascript file that requires the colorpicker used in the core (under the product attribute Visual Swatch in the admin panel)
````
// Vendor_Module/js/this/custom/file
require([
    'jquery',
    'jquery/colorpicker/js/colorpicker',
], function ($) {
    'use strict';
    // do anything, the damage is done
    console.log("required the colorpicker. Be free!");
````

### Expected result

1. Should see a message in the console that says "required the colorpicker. Be free!"
2. Can then customise the colorpicker as you would please.

### Actual result

1. Console error "Uncaught Error: Mismatched anonymous define() module" talking about the colorpicker.

To solve it, simply add any random string ID to the definition of the colorpicker.
Example
````
// lib/web/jquery/colorpicker/js/colorpicker.js:10
define("mage/colorpicker", [
	"jquery",
], function ($) {

````

A quick workaround is to simply repackage the colorpicker into the module/theme you want to use, with it patched as described, but it would be awesome to have this patched in the core, with the correct naming convention etc.

Please let me know if any more details are required.

**Comment by magento-engcom-team** [comment]
Hi @PivitParkour94. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me {$VERSION} instance` 

where `{$VERSION}` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (2.2-develop +). 
For more details, please, review the [Magento Contributor Assistant](https://github.com/magento/magento2/wiki/Magento-Contributor-Assistant) documentation.

@PivitParkour94 do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by PivitParkour94** [comment]
@magento-engcom-team give me 2.2.5 instance

**Comment by magento-engcom-team** [comment]
Hi @PivitParkour94. Thank you for your request. I'm working on Magento 2.2.5 instance for you

**Comment by magento-engcom-team** [comment]
Hi @PivitParkour94, here is your [Magento instance](https://i-17008-2-2-5.engcom.dev.magento.com).
Admin access: [https://i-17008-2-2-5.engcom.dev.magento.com/admin](https://i-17008-2-2-5.engcom.dev.magento.com/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by PivitParkour94** [comment]
This issue is a file bug, not a system config bug so I can't test it without ssh access or something, which I imagine is outside the possibilities of this automation.

Its super cool, but we'll probably have to get a human to look at this. I'll make a PR for the fix.

**Comment by PivitParkour94** [comment]
I should probably also say that you can't even bypass this by defining a path for the colorpicker in requirejs-config. The error is still being thrown without the string ID.

**Comment by engcom-backlog-nickolas** [comment]
Hello @PivitParkour94, thank you for your report.
We've acknowledged the issue and added to our backlog.

**Comment by engcom-backlog-nickolas** [comment]
@PivitParkour94, sorry, looks like I've not correctly reproduced this issue initially. 
Could you please provide more details?

**Comment by engcom-backlog-nickolas** [comment]
@PivitParkour94, I wasn't able to reproduce this issue by following the steps you provided, ‘jquery/colorpicker/js/colorpicker’ is available for me and there is no errors. If you'd like to update it, please reopen the issue.

**Comment by PivitParkour94** [comment]
I've noticed in core code that any reference to the `jquery/colorpicker/js/colorpicker` is inside the definition of a new js component i.e. `vendor/magento/module-swatches/view/adminhtml/web/js/visual.js:15` and `vendor/magento/module-swatches/view/adminhtml/web/js/form/element/swatch-visual.js:18`

However, I am wanting to include the library as is and run some anonymous code, using the quick javascript file given in the report
````
// Vendor_Module/js/this/custom/file
require([
    'jquery',
    'jquery/colorpicker/js/colorpicker',
], function ($) {
    'use strict';
    // do anything, the damage is done
    console.log("required the colorpicker. Be free!");
});
````
The colorpicker library isn't defined with an ID and so requiring the path is throwing a requirejs error.

Are you sure you aren't able to reproduce it when trying to `REQUIRE` the library? As opposed to use it in a new component with `DEFINE`.

If I should in fact be defining a new component for such a simple task, please let me know and I'll refactor accordingly.

Thanks
Nickolas

PS. I don't have permission to re-open this issue.
@engcom-backlog-nickolas

**Comment by tejasklevu** [comment]
it looks like this issue is not resolved in 2.4.x as well. can this be fixed?

**Comment by sibelius** [comment]
any workaround for this?
